OutOfSchool/OutOfSchool.WebApi/Controllers/V1/PictureController.cs

```diff
@@ -0,0 +1,323 @@
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PictureController"/> class.
+        /// </summary>
+        /// <param name="pictureService">Service for picture model.</param>
+        /// <param name="contentTypeProvider">Content type provider.</param>
+        /// <param name="config">Config.</param>
+        public PictureController(IPictureService pictureService, IContentTypeProvider contentTypeProvider, IConfiguration config)
+        {
+            this.pictureService = pictureService ?? throw new ArgumentNullException(nameof(pictureService));
+            this.contentTypeProvider = contentTypeProvider ?? throw new ArgumentNullException(nameof(contentTypeProvider));
+            this.maxPictureSize = config.GetValue<long>(KeyToMaxPictureSize);
+        }
+
+        /// <summary>
+        /// Get picture by it's id and workshop Id.
+        /// </summary>
+        /// <param name="workshopId">Id of the Workshop Entity.</param>
+        /// <param name="pictureId">Id of the picture.</param>
+        /// <returns>Workshop picture.</returns>
+        [AllowAnonymous]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(FileStreamResult))]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [HttpGet]
+        [Route("workshop/{workshopId}/picture/{pictureId}")]
+        public async Task<IActionResult> Workshop(long workshopId, Guid pictureId)
+        {
+            try
+            {
+                var pictureData = await pictureService.GetPictureWorkshop(workshopId, pictureId).ConfigureAwait(false);
+
+                return pictureData.ToFileResult();
+            }
+            catch (EntityNotFoundException)
+            {
+                return NotFound(pictureId);
+            }
+            catch
+            {
+                return BadRequest("An error occured while receiving the picture.");
+            }
+        }
+
+        /// <summary>
+        /// Get picture by it's id and provider Id.
+        /// </summary>
+        /// <param name="providerId">Id of the Provider Entity.</param>
+        /// <param name="pictureId">Id of the picture.</param>
+        /// <returns>Provider picture.</returns>
+        [AllowAnonymous]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(FileStreamResult))]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [HttpGet]
+        [Route("provider/{providerId}/picture/{pictureId}")]
+        public async Task<IActionResult> Provider(long providerId, Guid pictureId)
+        {
+            try
+            {
+                var pictureData = await pictureService.GetPictureProvider(providerId, pictureId).ConfigureAwait(false);
+
+                return pictureData.ToFileResult();
+            }
+            catch (EntityNotFoundException)
```

Comment:
Regarding exception handling - see https://itamoedreamteam.slack.com/archives/C01HGKDMSFQ/p1632908126132700 seems you might simplify your `try/catch`s in this Controller or remove them at all.